### PR TITLE
Use unique name for typed and untyped testprobes

### LINKED ISSUE #28509: Streams testkit - InvalidActorNameException when creating test probes
Hi!
I've managed to create a simple snippet that reproduces a bug

```
class FailingTest extends ScalaTestWithActorTestKit with AnyFreeSpecLike {

  "fails" in {
    implicit val untypedSystem: ActorSystem = testKit.system.toClassic
    implicit val materializer: Materializer = ActorMaterializer()

    val (pub, sub) = TestSource.probe[Int]
      .map(_ + 1)
      .toMat(TestSink.probe[Int])(Keep.both)
      .run()
    testKit.createTestProbe()
  }
}
```
The exception I'm getting is 
```
actor name [testProbe-1] is not unique!
akka.actor.InvalidActorNameException: actor name [testProbe-1] is not unique!
	at akka.actor.dungeon.ChildrenContainer$NormalChildrenContainer.reserve(ChildrenContainer.scala:134)
	at akka.actor.dungeon.Children.reserveChild(Children.scala:139)
	at akka.actor.dungeon.Children.reserveChild$(Children.scala:137)
	at akka.actor.ActorCell.reserveChild(ActorCell.scala:447)
	at akka.actor.dungeon.Children.makeChild(Children.scala:289)
	at akka.actor.dungeon.Children.attachChild(Children.scala:51)
	at akka.actor.dungeon.Children.attachChild$(Children.scala:50)
	at akka.actor.ActorCell.attachChild(ActorCell.scala:447)
	at akka.actor.ActorSystemImpl.systemActorOf(ActorSystem.scala:774)
	at akka.testkit.TestKitBase.$init$(TestKit.scala:181)
	at akka.testkit.TestKit.<init>(TestKit.scala:923)
	at akka.testkit.TestProbe.<init>(TestKit.scala:984)
	at akka.testkit.TestProbe.<init>(TestKit.scala:986)
	at akka.testkit.TestProbe$.apply(TestKit.scala:1020)
	at akka.stream.testkit.TestSubscriber$ManualProbe.<init>(StreamTestKit.scala:298)
	at akka.stream.testkit.TestSubscriber$Probe.<init>(StreamTestKit.scala:758)
	at akka.stream.testkit.TestSubscriber$.probe(StreamTestKit.scala:286)
	at akka.stream.testkit.StreamTestKit$ProbeSink.create(StreamTestKit.scala:860)
	at akka.stream.impl.SinkModulePhase.materializeAtomic(PhasedFusingActorMaterializer.scala:870)
	at akka.stream.impl.PhasedFusingActorMaterializer.materialize(PhasedFusingActorMaterializer.scala:492)
	at akka.stream.impl.PhasedFusingActorMaterializer.materialize(PhasedFusingActorMaterializer.scala:451)
	at akka.stream.impl.PhasedFusingActorMaterializer.materialize(PhasedFusingActorMaterializer.scala:444)
	at akka.stream.scaladsl.RunnableGraph.run(Flow.scala:629)
	at typeracer.flow.FailingTest.$anonfun$new$1(Test.scala:21)
```
Creating the probe first throws the same exception.

Is this an expected behavior? What I'm missing?

Thanks!

PS: `akka-actor-typed`, `akka-actor-testkit-typed` and `akka-stream-testkit` are at version 2.5.26

### PR DESCRIPTION
**Pull request by chbatey**
Fixes https://github.com/akka/akka/issues/28509

### CODE REVIEW

File: akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala [hunk 1]
```diff
@@ -85,6 +85,13 @@ class ActorTestKitSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike wi
       scalaTestWithActorTestKit2.testKit.system.settings.config.hasPath("test.from-application") should ===(false)
     }
 
+    "have unique names for probes across untyped testkit" in {
+      import akka.actor.typed.scaladsl.adapter._
+      createTestProbe()
+      akka.testkit.TestProbe()(system.toClassic)
+      // not throw
```
Comment (johanandren) [inline akka-actor-testkit-typed/src/test/scala/akka/actor/testkit/typed/scaladsl/ActorTestKitSpec.scala]: best comment ever

**Review by johanandren — APPROVED** [review]

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala [hunk 2]
```diff
@@ -36,8 +36,6 @@ import akka.util.PrettyDuration._
 
 @InternalApi
 private[akka] object TestProbeImpl {
-  private val testActorId = new AtomicInteger(0)
```
Comment (patriknw) [inline akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala]: I think that import is too magic/hidden. Would prefer 
```
// have to use same global counter as Classic TestKit to ensure unique names
private def testActorId = akka.testkit.TestKit.testActorId
```

**Review by patriknw — COMMENTED** [review]
small thing...

File: akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala (hunk 2 above)
Comment (chbatey) [reply akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/TestProbeImpl.scala]: Okay

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by renatocaval** [comment]
Agian, issue with Jabba in Travis. Re-started.

**Pull request merged**